viewer/src/datamodels/cad/styling/NodeAppearanceTextureBuilder.ts

```diff
@@ -0,0 +1,281 @@
+  }
+
+  getDefaultAppearance(): NodeAppearance {
+    return this._defaultAppearance;
+  }
+
+  /**
+   * Sets the default style and invalidates the builder. Note that this causes a full
+   * recomputation of values the next time {@link build} is called, so using this might be
+   * expensive.
+   * @param appearance New style that is applied to all 'unstyled' elements.
+   */
+  setDefaultAppearance(appearance: NodeAppearance) {
+    if (!equalNodeAppearances(appearance, this._defaultAppearance)) {
+      this._defaultAppearance = appearance;
+      fillColorTexture(this._overrideColorPerTreeIndexTexture, appearance);
+      this._infrontNodesTreeIndices.clear();
+      this._ghostedNodesTreeIndices.clear();
+      this._regularNodesTreeIndices.clear();
+
+      const allIndicesRange = new NumericRange(0, this._treeIndexCount);
+      const infront = !!appearance.renderInFront;
+      const ghosted = !!appearance.renderGhosted;
+      if (infront) {
+        this._infrontNodesTreeIndices.addRange(allIndicesRange);
+      } else if (ghosted) {
+        this._ghostedNodesTreeIndices.addRange(allIndicesRange);
+      } else {
+        this._regularNodesTreeIndices.addRange(allIndicesRange);
+      }
+
+      // Force full update as we might have overwritten previously applied styles
+      this._currentlyAppliedStyles.clear();
+      this._needsUpdate = true;
+    }
+  }
+
+  get regularNodeTreeIndices(): IndexSet {
+    return this._regularNodesTreeIndices;
+  }
+
+  get ghostedNodeTreeIndices(): IndexSet {
+    return this._ghostedNodesTreeIndices;
+  }
+
+  get infrontNodeTreeIndices(): IndexSet {
+    return this._infrontNodesTreeIndices;
+  }
+
+  get needsUpdate(): boolean {
+    return this._needsUpdate;
+  }
+
+  /**
+   * A texture holding at least one element per node with color override
+   * and style information. RGB is used to store color, A is used to store
+   * style toggles, with the following bit layout:
+   * - 0  : visible bit   - when set the node is visible
+   * - 1  : in front bit  - when set the node is rendered in front of other objects
+   * - 2  : ghosted bit   - when set the node is rendered 'ghosted'
+   * - 3-5: outline color - outline toggle and color ({@see OutlineColor}).
+   * - 6-7: unused
+   * Note that in-front and ghost information also is available from
+   * the {@see inFrontTreeIndices} and {@see ghostedTreeIndices} collections.
+   */
+  get overrideColorPerTreeIndexTexture(): THREE.DataTexture {
+    return this._overrideColorPerTreeIndexTexture;
+  }
+
+  dispose() {
+    this._styleProvider.off('changed', this._handleStylesChangedListener);
+    this._overrideColorPerTreeIndexTexture.dispose();
+  }
+
+  build() {
+    const orphanStyleIds = new Set<number>(this._currentlyAppliedStyles.keys());
+    // TODO 2021-02-04 larsmoa: Currently transform overrides are never removed
+
+    // 1. Reset nodes that has been removed from nodesets to default style
```

Comment:
If you have to tag comments with 1. -> 2. -> 3. etc, then they should probably be in its own functions, since they clearly have some logical separation.

Reply:
Probably. The TODO seems very important though :|